.github/workflows/windows_and_macOS.yml

```diff
@@ -0,0 +1,39 @@
+name: win-mac-ci
+on:
+  push:
+    branches:
+      - master
+  pull_request: {}
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: ['windows-latest', 'macOS-latest']
+        ghc: '9.0'
+    steps:
+    - uses: actions/checkout@v2
+    - uses: haskell/actions/setup@v1
+      id: setup-haskell-cabal
+      with:
+        ghc-version: ${{ matrix.ghc }}
+    - name: Update cabal package database
+      run: cabal update
+    - uses: actions/cache@v2.1.4
+      name: Cache cabal stuff
+      with:
+        path: ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
+        key: ${{ runner.os }}-${{ matrix.ghc }}
+    - name: Test
+      run: cabal test
+    - name: Bench
+      run: cabal bench
```

Comment:
Is it good idea to **run** benchmarks on CI? The `text` benchmark suite isn't very fast to run (and need benchmark dta checkout too, IIRC)

Reply:
I think `cabal bench --benchmark-option=-l` would be good enough, just to check that benchmarks are still buildable under Windows/MacOS.

Reply:
I started by copypasting bytestring's CI config, I agree that running `cabal bench` for text is too much.